src/backend/joanie/core/models/products.py

```diff
@@ -435,6 +435,45 @@ def __str__(self):
         active = _("active") if self.is_active else _("inactive")
         return f"[{active}][{self.state}] {self.user} for {self.course_run}"
 
+    @property
+    def grade_cache_key(self):
+        """The cache key used to store enrollment's grade."""
+        return f"grade_{self.uid}"
+
+    @property
+    def is_passed(self):
+        """Get enrollment grade then return the `passed` property value or False"""
+        grade = self.get_grade()
+
+        return grade["passed"] if grade else False
+
+    def get_grade(self):
+        """Retrieve the grade from the related LMS then store result in cache."""
+        grade = cache.get(self.grade_cache_key)
+
+        if grade is None:
+            lms = LMSHandler.select_lms(self.course_run.resource_link)
+
+            if lms is None:
+                logger.error(
+                    "Course run %s has no related lms.",
+                    self.course_run.id,
+                )
+            else:
+                try:
+                    grade = lms.get_grades(
+                        username=self.user.username,
+                        resource_link=self.course_run.resource_link,
+                    )
+                except GradeError:
+                    pass
+
+            cache.set(
+                self.grade_cache_key, grade, settings.JOANIE_ENROLLMENT_GRADE_CACHE_TTL
+            )
```

Comment:
I would say you should only set cache when the grade was successfully retrieved:
```suggestion
                try:
                    grade = lms.get_grades(
                        username=self.user.username,
                        resource_link=self.course_run.resource_link,
                    )
                except GradeError:
                    pass
                else:
                    cache.set(
                        self.grade_cache_key, grade, settings.JOANIE_ENROLLMENT_GRADE_CACHE_TTL
                    )
```